### PR TITLE
perf: move serialization to background threads

### PR DESCRIPTION
**Pull request by hassiebp**
> [!IMPORTANT]
> Move serialization to background threads, improve logging, and add performance overhead test for Langfuse tracing.
> 
>   - **Behavior**:
>     - Move serialization to background threads in `langfuse/task_manager.py`.
>     - Handle serialization errors in `Consumer._next()` in `langfuse/task_manager.py`.
>     - Add `_filter_io_from_event_body()` to filter `input` and `output` from logs in `langfuse/client.py`.
>   - **Logging**:
>     - Update logging to exclude `input` and `output` in `langfuse/client.py` and `langfuse/task_manager.py`.
>     - Add debug logs for batch processing in `Consumer` class in `langfuse/task_manager.py`.
>   - **Testing**:
>     - Add `test_langfuse_overhead()` in `tests/test_langchain.py` to measure performance overhead of Langfuse tracing.
>   - **Misc**:
>     - Remove unused `kwargs_log` in `_log_debug_event()` in `langfuse/callback/langchain.py`.
> 
> <sup>This description was created by </sup>[Ellipsis](https://www.ellipsis.dev?ref=langfuse%2Flangfuse-python&utm_source=github&utm_medium=referral)<sup> for 3a22aeedfde229c376bf2440816ea4e4c7598e1e. It will automatically update as commits are pushed.</sup>

### CODE REVIEW

File: langfuse/callback/langchain.py [hunk 1]
```diff
@@ -991,14 +991,8 @@ def _log_debug_event(
...
         **kwargs,
     ):
-        kwargs_log = (
-            ", " + ", ".join([f"{key}: {value}" for key, value in kwargs.items()])
-            if len(kwargs) > 0
-            else ""
-        )
         self.log.debug(
             f"Event: {event_name}, run_id: {str(run_id)[:5]}, parent_run_id: {str(parent_run_id)[:5]}"
-            + kwargs_log
```
Comment (maxdeichmann) [inline langfuse/callback/langchain.py:1001]: Why did you remove this?

File: langfuse/client.py [hunk 2]
```diff
@@ -3350,3 +3360,7 @@ def __init__(self, dataset: Dataset, items: typing.List[DatasetItemClient]):
         self.created_at = dataset.created_at
         self.updated_at = dataset.updated_at
         self.items = items
+
+
+def _filter_io_from_event_body(event_body: Dict[str, Any]):
+    return {k: v for k, v in event_body.items() if k not in ("input", "output")}
```
Comment (maxdeichmann) [inline langfuse/client.py:3366]: i would also remove metadata here.

File: langfuse/task_manager.py [hunk 3]
```diff
@@ -107,13 +113,37 @@ def _next(self):
             try:
                 item = queue.get(block=True, timeout=self._flush_interval - elapsed)
 
+                # convert pydantic models to dicts
+                if "body" in item and isinstance(item["body"], pydantic.BaseModel):
+                    item["body"] = item["body"].dict()
```
Comment (maxdeichmann) [inline langfuse/task_manager.py:118]: exclude_none=True is missing here i think. I think we should only send diffs to the backend.

File: langfuse/callback/langchain.py (hunk 1 above)
Comment (hassiebp) [reply langfuse/callback/langchain.py:1001]: expensive calculation, added overhead in multiple hundreds of milliseconds for large objects
